testdata/generate.py

```diff
@@ -0,0 +1,526 @@
+        for x in range(num_contacts):
+            contacts.append(contact_gen.make_one(account_id=id))
+        # name household based on contacts
+        attrs["name"] = self.name_household(contacts)
+
+        # addresses (some seasonal)
+        address_gen = self.generators["addresses"]
+        num_addresses = address_gen.choose_count()
+        if num_addresses == 1:
+            address_gen.make_one(household_id=id)
+        else:
+            address_gen.make_one(
+                household_id=id, start_month=6, start_day=1, end_month=8, end_day=31
+            )
+            address_gen.make_one(
+                household_id=id, start_month=9, start_day=1, end_month=5, end_day=31
+            )
+
+        # recurring donations
+        rd_gen = self.generators["recurring_donations"]
+        num_rds = rd_gen.choose_count()
+        if num_rds:
+            rd_gen.make_one(attrs, contacts)
+
+        # one-off opportunities
+        opp_gen = self.generators["opportunities"]
+        if self.include_skew and n == 1:
+            num_opps = 50000
+            attrs["name"] = "50k Opp Household"
+        elif self.include_skew and n < 10:
+            num_opps = 5000
+            attrs["name"] = "5k Opp Household"
+        else:
+            num_opps = opp_gen.choose_count()
+        for x in range(num_opps):
+            opp_gen.make_one(contacts, account_id=id)
+
+        self.writer.writerow(attrs)
+        return attrs
+
+    def make_many(self, count, **kw):
+        i = 0
+        while i < count:
+            i += 1
+            self.make_one(i, **kw)
+            if not i % 1000:
+                self.logger.info(i)
+
+    def name_household(self, children):
+        return "{} Household".format(
+            "/".join(set(contact["last_name"] for contact in children))
+        )
+
+
+class GenerateFakeData(BaseTask):
+
+    task_options = {
+        "count": {
+            "description": "How many households to generate. Related objects will be generated proportionally.",
+            "required": True,
+        },
+        "database_url": {
+            "description": "The database url for an empty postgres database to write data to.",
+            "required": True,
+        },
+        "include_skew": {
+            "description": "If true, include a few households with thousands of opportunities.",
+            "required": False,
+        },
+    }
+
+    def _init_options(self, kw):
+        super(GenerateFakeData, self)._init_options(kw)
+        self.options["count"] = int(self.options["count"])
+        self.options["include_skew"] = process_bool_arg(
+            self.options.get("include_skew")
+        )
+
+    def _run_task(self):
+        database_url = self.options["database_url"]
+        count = self.options["count"]
+
+        # patch faker for performance
+        self._install_faker_patch()
+
+        self.logger.info("Connecting to {}".format(database_url))
+        conn = psycopg2.connect(database_url)
```

Comment:
Is it possible that we could just use sqlalchemy to connect to the database here instead of calling psycopg2 directly?  I'd like to test this with generating data to a sqlite database if possible which seems like a common use case.

Reply:
Inserting from CSV using COPY FROM is only supported by postgres, but I may take a stab at supporting other dbs the way I did for the bulkdata tasks. It'd be nice to have that flexibility.